File: Phind-CodeLlama-34B-v2/RAG-template/declare_directive_device_resident_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:declare directive device_resident clause,V:2.7-3.3
int test1(){
    int err = 0;
    int *a, *b, *c, *d, *e, *f;

    a = (int *)malloc(n * sizeof(int));
    b = (int *)malloc(n * sizeof(int));
    c = (int *)malloc(n * sizeof(int));
    d = (int *)malloc(n * sizeof(int));
    e = (int *)malloc(n * sizeof(int));
    f = (int *)malloc(n * sizeof(int));

    #pragma acc declare device_resident(a, b, c)

    #pragma acc data copy(a[0:n], b[0:n], c[0:n], d[0:n], e[0:n], f[0:n])
    {
        #pragma acc parallel present(a, b, c, d, e, f)
        {
            for (int x = 0; x < n; ++x){
                a[x] = b[x] + c[x];
                d[x] = a[x] * e[x];
                f[x] = d[x] - a[x];
            }
        }
    }

    for (int x = 0; x < n; ++x){
        if (f[x] != (b[x] + c[x]) * e[x] - (b[x] + c[x])){
            err = 1;
        }
    }

    free(a);
    free(b);
    free(c);
    free(d);
    free(e);
    free(f);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/declare_directive_device_resident_clause.f90
!T1:declare directive device_resident clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, DIMENSION(:), ALLOCATABLE :: a
        INTEGER, DIMENSION(:), ALLOCATABLE :: a_dev
        INTEGER :: i

        ! Allocate host memory
        ALLOCATE(a(10))
        a = 1

        !$acc data copyin(a) create(a_dev)
        !$acc declare create(a_dev) device_resident(a_dev)

        !$acc parallel loop present(a, a_dev)
        DO i = 1, 10
          a_dev(i) = a(i)
        END DO

        !$acc end data

        ! Check if the data is in device memory
        !$acc data copy(a_dev)
        !$acc parallel loop present(a_dev)
        DO i = 1, 10
          IF (a_dev(i) .ne. 1) THEN
            errors = errors + 1
          END IF
        END DO
        !$acc end data

        ! Deallocate host memory
        DEALLOCATE(a)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM